### PR TITLE
Update developer section, add Filter API tutorials

### PR DESCRIPTION
**Pull request by nlsdvl**
This PR updates the developer section, top level page, navigation tree, and adds a tutorial section. 
This tutorial section focuses on the Filter API usage, whereas the HOWTOs section is focused on using GPAC and MP4Box exectuables.

### CODE REVIEW

File: docs/Developers/javascript.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+Javascript developers have two options to work with GPAC, the most appropriate solution depends on the project's goal:
+
+## Javascript Filters
+
+The JS filter API makes it easy to **extend gpac** using the internal QuickJS runtime, giving access to the Filter API for frame and packet processing, but also APIs foradaptative streaming, compositing, storage, ...
```
Comment (rbouqueau) [inline docs/Developers/javascript.md:6]: foradaptive -> for adaptive (missing space)

File: docs/Developers/tutorials/GPAC-concepts.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+
+## What is GPAC ?
+
+The **GPAC Filter API** is at the core of the [MP4Box and GPAC](Howtos/gpac-mp4box) applications.
+
+The `gpac` application allows building media pipeplines by conveniently [combining and configuring Filters](Filters/filters_general) from the command line.
```
Comment (rbouqueau) [inline docs/Developers/tutorials/GPAC-concepts.md:6]: pipeplines => pipelines

**Comment by rbouqueau** [comment]
Perfect (aside from 2 typos), thank you!

**Review by jeanlf — APPROVED** [review]
Looks good enough for a starting point, thanks !

**Comment by rbouqueau** [comment]
I fixed my own typos on master ;)

**Pull request merged**